src/plugins/mermaid.js

```diff
@@ -0,0 +1,26 @@
+import { visit } from 'unist-util-visit';
```

Comment:
This is a rehype plugin that replaces all code blocks with marked with the `mermaid` language with a `<div class="mermaid">`

The presence of `class="mermaid" allows the mermaid renderer to find them and transform the code to a mermaid diagram

The gold standard here would be to generate mermaid diagrams at build time but this can be a future enhancement